src/Entities/RegistryContext.h

```diff
@@ -0,0 +1,45 @@
+#include <Graphics/DebugLines.h>
+
+namespace openblack::Entities {
+class RenderContext
+{
+public:
+	std::unique_ptr<graphics::DebugLines> streams;
+	std::unique_ptr<graphics::DebugLines> boundingBox;
+
+	struct InstancedDrawDesc
+	{
+		InstancedDrawDesc(uint32_t offset, uint32_t count) : offset(offset), count(count) {}
+		uint32_t offset;
+		uint32_t count;
+	};
+
+	/// A list of cpu-side uniforms which is refilled at every \ref PrepareDraw.
+	/// This vector will resize to the number of instances it manages
+	/// but in practice, it should only grow its reserved memory.
+	/// If debug bounding boxes are enabled, it will double in size to fit all
+	/// bounding boxes in the second half of the list.
+	std::vector<glm::mat4> instanceUniforms;
+	/// Stores information for rendering which is prepared at \ref PrepareDraw.
+	std::map<MeshId, const InstancedDrawDesc> instancedDrawDescs;
+	/// Not an actual vertex buffer, but a dynamic general purpose buffer which
+	/// stores uniform data as a GPU-side copy of \ref _instanceUniforms and
+	/// which is populated in \ref PrepareDraw and consumed in \ref DrawModels.
+	/// This buffer will resize if the size of \ref _instanceUniforms exceeds
+	/// its allocated size. It will never shrink.
+	/// The values stored are a list of uniforms (model matrix) needed for both
+	/// the instances of entities and their bounding boxes.
+	bgfx::DynamicVertexBufferHandle instanceUniformBuffer;
```

Comment:
These need to be destroyed with bgfx when the class is destructed.
See https://github.com/openblack/openblack/pull/150/files#diff-b3ae7df02339c5d79de929ff2f8401ecL30-L31

Otherwise you get leaks when you shut down:
```
 BGFX Shutdown...
 BGFX render thread exit
 BGFX LEAK: DynamicVertexBufferHandle 1 (max: 4096)
 BGFX 	  0:    0
 BGFX LEAK: VertexBufferHandle 1 (max: 4096)
 BGFX 	  0: 1357 
 BGFX Shutdown complete.
```

Reply:
This is done when the registry is cleared

Reply:
It needs to be done while shutting down too. Otherwise you get those warnings.

Reply:
Sorted